### ISSUE TITLE
Ways to test integrity of an image

### ISSUE THREAD

**Issue by jdiversen** [OpenImageIO/oiio#1061]
Hi there,

If this is not possible already, could there possibly be a a couple of new convenience features added that might make it easy to test the integrity of an image (reading each scanline or tile), and return information on failed reads?

I would love there there to be python convenience method for this (an ImageInput.isComplete() and ImageInput.testIntegrity()), as well as a corresponding oiiotool command which simply prints the formatted results to stdout.

I'm currently using 1.2.1 getting some odd results with a partially completed EXR. For the valid scanlines I'm getting a "Unexpected end of file." and the missing scanlines I'm getting the correct "Scan line ### is missing."

My code is currently:
```python
import OpenImageIO as oiio
import array

i = oiio.ImageInput.open("/var/tmp/resumetest.0048.exr")
spec = i.spec()
data = array.array("B", "1234" * spec.scanline_bytes(True))

if spec.tile_width == 0:
    for y in range(spec.y, spec.y+spec.height) :
        print y, 
        if not i.read_scanline(y, spec.z, data):
            print i.geterror()
```

**Comment by jdiversen** [comment]
I have since upgraded to 1.5.12, and am experiencing further issues.  Any attempt to read data from the file is returning None from read_scanline().

In the following code:

```python
import OpenImageIO as oiio
import array
import sys
import re 

if len(sys.argv) != 2 :
    print "Need 1 arguments: inputfilename"
    exit(1)

filename = sys.argv[1]
input = oiio.ImageInput.open(filename)
spec = input.spec()
if spec.tile_width == 0 :
    for y in range(spec.y, spec.y+spec.height) :
        pixels = input.read_scanline (y, spec.z, oiio.UNKNOWN)
        if pixels == None :
            print "Broken at scanline " + str(y)
            exit(1)

print "File " + filename + " is ok"
```
I get a "broken scanline" no matter what I try - the return value for pixels is None. Running an input.geterror() continues to report an "Unexpected end of file" for scanlines which are readable in the C++ API.

I am forbidden from attaching my test EXR to this bug.

**Comment by lgritz** [comment]
I can't reproduce this.  I used your script above, with two additions:

1. Add at the top

    #!/usr/bin/env python

2. Add right after both of your 'print' lines:

    print "geterr = " + input.geterror()

With a freshly built 1.5.12 (and also master), I get these results:

    $ ./iverson1.py good.exr
    File good.exr is ok
    geterr =

    $ ./iverson1.py incomplete.exr
    Broken at scanline 416
    geterr = Failed OpenEXR read: Error reading pixel data from image file "incomplete.exr". Scan line 416 is missing.

This seems like the expected behavior.

Are you quite sure that you are picking up the right OIIO version (including python library)?

**Comment by jdiversen** [comment]
Larry narrowed this down to an issue in OpenEXR-2.0.1 subsequently fixed in 2.1.0+
